xbmc/video/VideoThumbLoader.cpp

```diff
@@ -349,6 +351,21 @@ bool CVideoThumbLoader::LoadItemCached(CFileItem* pItem)
     SetArt(*pItem, artwork);
   }
 
+  // hide thumb if episode is unwatched 
+  std::shared_ptr<CSettingList> setting(std::dynamic_pointer_cast<CSettingList>(
+    CServiceBroker::GetSettingsComponent()->GetSettings()->GetSetting(CSettings::SETTING_VIDEOLIBRARY_SHOWUNWATCHEDPLOTS)));
+  if (pItem->HasArt("thumb") && pItem->HasVideoInfoTag() &&
+      pItem->GetVideoInfoTag()->m_type == MediaTypeEpisode &&
+      pItem->GetVideoInfoTag()->GetPlayCount() == 0 &&
+      setting && 
+      !setting->FindIntInList(CSettings::VIDEOLIBRARY_THUMB_SHOW_UNWATCHED_EPISODE)
+     )
+  {
+    pItem->SetArt("thumb", "OverlaySpoiler.png");
+    m_videoDatabase->Close();
+    return true;
```

Comment:
The exact same code is already after the if block.

Reply:
Wasn't sure, if I need to close the database. Will test removing that and update the PR if everything works fine (not arguing that it won't ;) ).

Reply:
have look below, database is closed there too